ec2/main.tf

```diff
@@ -0,0 +1,136 @@
+#Create Security Groups for public instances
+resource "aws_security_group" "public_ec2_sg" {
+  name = "public_ec2_sg"
+  description = "Allows TLS inbound traffic and outbound traffic"
+  vpc_id = aws_vpc.practice_vpc.id
+
+/*
+By default, AWS creates an ALLOW ALL egress rule when creating 
+a new Security Group inside of a VPC. When creating a new 
+Security Group inside a VPC, Terraform will remove this default 
+rule, and require you specifically re-create it if you desire 
+that rule. We feel this leads to fewer surprises in terms of 
+controlling your egress rules. If you desire this rule to be in 
+place, you can use this egress block:
+*/
+
+  egress {
```

Comment:
This will force a destroy of the security group itself if we have changes on the ingress or egress , its better to use the security group rule resource
see ref: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule.html

Reply:
Yes, I got that, but the docs also states AWS does a default allow all egress when creating a security group, but terraform won't do it if not explicitly stated.

On a second though, this is the same as the aws_vpc_security_egress_rule resource I created. I will comment it out.

Thank you.

Reply:
Exactly, you created that already from another resource, hence not needed in the security group resource